src/app.jsx

```diff
@@ -420,14 +417,14 @@ class Application extends React.Component {
             this.updateContainerAfterEvent(event.Actor.ID, system, event);
             break;
         case 'remove':
-            // HACK: we don't get a pod event when a container in a pod is removed.
-            // https://github.com/containers/podman/issues/15408
-            if (event.Actor.Attributes.podId) {
-                this.updatePodAfterEvent(event.Actor.Attributes.podId, system);
-            } else {
-                this.updatePodsAfterEvent(system);
-            }
-            this.updateContainersAfterEvent(system);
+            this.updateContainersAfterEvent(system).then(() => {
+                // HACK: we don't get a pod event when a container in a pod is removed.
+                // https://github.com/containers/podman/issues/15408
+                if (event.Actor.Attributes.podId)
+                    this.updatePodAfterEvent(event.Actor.Attributes.podId, system);
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1324-20230704-180519-dee6ebfb-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)